src/katsdpmodels/models.py

```diff
@@ -207,6 +220,43 @@ def from_file(cls: Type[_H], file: io.IOBase, url: str, *,
                 else:
                     return model
 
+    def to_file(self, file: Union[str, pathlib.Path, _FileLike], *,
+                content_type: Optional[str] = None) -> None:
+        if self.version is None:
+            raise ValueError('Version must be set before writing file')
+        if content_type is not None:
+            if content_type != 'application/x-hdf5':
+                raise FileTypeError(f'Expected application/x-hdf5, not {content_type}')
+        else:
+            if isinstance(file, (str, pathlib.Path)):
+                path = pathlib.Path(file)
+            else:
+                path = pathlib.Path(file.name)   # type: ignore
+            if path.suffix not in {'.h5', '.hdf5'}:
```

Comment:
Should we check for existence of the file and warn about overwrite ?

Reply:
I've never been a fan of that. It's then a pain to avoid the warning if you don't want it, and it's racy. But if you like I can add a kwarg to open the file in exclusive mode so that you get an OSError if it already exists.

Reply:
I know, but I still prefer not clobbering without at least a docstring warning, or a mode kwarg (defaulting to exclusive).

Reply:
I've added a note to the docstring.